### PR TITLE
2.5.2 release notes

### PR DESCRIPTION
**Pull request by nathan-px**

### CODE REVIEW

File: content/reference/release-notes/portworx.md [hunk 1]
```diff
@@ -6,6 +6,48 @@ keywords: portworx, release notes
...
+| PWX-13171 | The `px_volume_readthroughput`, `px_volume_writethrougput` and `px_volume_iops` metrics did not update.<br/><br/>**User Impact:** Users may have seen values for these metrics reported as zero.<br/><br/>**Resolution:** Portworx once again updates these metrics. <!-- does this need an RN? -->| 
+| PWX-12088 | Portworx used a version of IBM KeyProtect that caused a kernel panic if multiple threads tried to use it.<br/><br/>**User Impact:** Portworx nodes experiencing a kernel panic restarted, and some apps did not come back online after recovery.<br/><br/>**Resolution:** Portworx now uses IBM KeyProtect library v0.3.5, which solves this problem. |
+
+### Known Issues
+
+Portworx is aware of the following issues, check future release notes for fixes on these issues:
+
+|**Issue Number**|**Issue Description**|**Workaround**|
+|----|----|----|
+| PWX-13127 | Known issue in 2.5.2 |
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:49]: Does this need an RN, or can it be removed?

File: content/reference/release-notes/portworx.md [hunk 2]
```diff
@@ -6,6 +6,48 @@ keywords: portworx, release notes
...
+
+Portworx has upgraded or enhanced functionality in the following areas:
+
+| **Improvement Number** | **Improvement Description** |
+|----|----|
+| PWX-12737 | Added support for automatic cloud disk management for non-public Azure clouds like US Government, Germany and China clouds. |
+| PWX-13082 | You can now configure the frequency with which Portworx takes its KVDB backups using the `kvdb_dump_frequency_minutes` runtime option. |
+| PWX-10216 | Added support for Vault Namespaces. |
+| PWX-12466 | The `--all` option for the `pxctl cloudmigrate start` CLI command has been deprecated. | 
+| PWX-12603 | By default when sharedv4 volumes are in use, PX will use 16 threads to process them. You can change the total number of threads in use by PX by running "pxctl cluster options update --sharedv4-threads 32" <!-- is this an improvement? --> |
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:29]: @adityadani is this an improvement? There's not much info in the ticket to go by.

File: content/reference/release-notes/portworx.md [hunk 3]
```diff
@@ -6,6 +6,48 @@ keywords: portworx, release notes
...
 
+## 2.5.2
+
+May 29, 2020
+
+### New features
+
+ * Introducing [Vault](): .
+
+ <!-- need more info on vault changes -->
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:17]: @adityadani Does the vault doc change warrant a new feature notice, or is the `| PWX-10216 | Added support for Vault Namespaces. |` line under improvements the same thing?

File: content/reference/release-notes/portworx.md [hunk 4]
```diff
@@ -6,6 +6,48 @@ keywords: portworx, release notes
...
+| PWX-12827 | When running lengthy `pxctl` operations, Portworx timed-out and restarted. <br/><br/>**User Impact:** Users were unable to successfully run `pxctl` operations if they took too long to execute<br/><br/>**Resolution:** Execution timeout can be increased through `pxctl` cluster options. |
+| PWX-12655 | When Portworx images were uploaded to nodes as tarballs (e.g., through the `docker save` or `ssh node docker load` commands), Docker may not have set the image digest properly.<br/><br/>**User Impact:** When the image digest was not available, OCI-Monitor would not detect manually uploaded Portworx images, and would attempt to pull the Portworx image, potentially failing in air-gapped environments.<br/><br/>**Resolution:** Portworx now has improved image detection, even in cases where image digests are not available. | 
+| PWX-13171 | The `px_volume_readthroughput`, `px_volume_writethrougput` and `px_volume_iops` metrics did not update.<br/><br/>**User Impact:** Users may have seen values for these metrics reported as zero.<br/><br/>**Resolution:** Portworx once again updates these metrics. <!-- does this need an RN? -->|
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:40]: @vilasd I wasn't sure if this needed a note or not, so I added one. should we keep it?

File: content/reference/release-notes/portworx.md [hunk 5]
```diff
@@ -6,6 +6,48 @@ keywords: portworx, release notes
...
+| PWX-12827 | When running lengthy `pxctl` operations, Portworx timed-out and restarted. <br/><br/>**User Impact:** Users were unable to successfully run `pxctl` operations if they took too long to execute<br/><br/>**Resolution:** Execution timeout can be increased through `pxctl` cluster options. |
+| PWX-12655 | When Portworx images were uploaded to nodes as tarballs (e.g., through the `docker save` or `ssh node docker load` commands), Docker may not have set the image digest properly.<br/><br/>**User Impact:** When the image digest was not available, OCI-Monitor would not detect manually uploaded Portworx images, and would attempt to pull the Portworx image, potentially failing in air-gapped environments.<br/><br/>**Resolution:** Portworx now has improved image detection, even in cases where image digests are not available. |
```
Comment (zoxpx) [inline content/reference/release-notes/portworx.md:39]: ~through the `docker save` or `ssh node docker load` commands~
* it's not `or` -- the actual command was `docker save portworx/px-enterprise:xxx | ssh node123 docker load`
* suggest re-wording as:
> When Portworx images were uploaded to nodes via `docker load` command, Docker may not have set the image digest properly ...

File: content/reference/release-notes/portworx.md (hunk 2 above)
Comment (adityadani) [reply content/reference/release-notes/portworx.md:29]: Yes this is an improvement. Should we change this to:
```
When sharedv4 volumes are in use, PX by default uses 16 NFS threads to process them. You can change the total number of threads used by PX by running the "pxctl cluster options update --sharedv4-threads <num>"
```

File: content/reference/release-notes/portworx.md [hunk 6]
```diff
@@ -6,6 +6,48 @@ keywords: portworx, release notes
...
+### Improvements
+
+Portworx has upgraded or enhanced functionality in the following areas:
+
+| **Improvement Number** | **Improvement Description** |
+|----|----|
+| PWX-12737 | Added support for automatic cloud disk management for non-public Azure clouds like US Government, Germany and China clouds. |
+| PWX-13082 | You can now configure the frequency with which Portworx takes its KVDB backups using the `kvdb_dump_frequency_minutes` runtime option. |
+| PWX-10216 | Added support for Vault Namespaces. |
+| PWX-12466 | The `--all` option for the `pxctl cloudmigrate start` CLI command has been deprecated. |
```
Comment (vilasd) [inline content/reference/release-notes/portworx.md:28]: This is not improvement. Move it to fixes.

File: content/reference/release-notes/portworx.md (hunk 4 above)
Comment (vilasd) [reply content/reference/release-notes/portworx.md:40]: ok to keep it

File: content/reference/release-notes/portworx.md (hunk 2 above)
Comment (adityadani) [reply content/reference/release-notes/portworx.md:29]: We are also missing PWX-11288:  Added support for Vault Kubernetes Authentication.
We are making doc changes for this as well - https://github.com/portworx/pxdocs/pull/814

File: content/reference/release-notes/portworx.md (hunk 3 above)
Comment (adityadani) [reply content/reference/release-notes/portworx.md:17]: Lets keep that as an improvement. And yes we need a doc for it. Will work on it today.

File: content/reference/release-notes/portworx.md (hunk 1 above)
Comment (vilasd) [reply content/reference/release-notes/portworx.md:49]: We don't need to

**Comment by vilasd** [comment]
Missed release notes for this PWX-12512

File: content/reference/release-notes/portworx.md [hunk 7]
```diff
@@ -6,6 +6,36 @@ keywords: portworx, release notes
...
+| PWX-12512 | PX-Essential can now refresh licenses through an HTTP or HTTPS proxy. |
+
+### Fixes
+
+The following issues have been fixed:
+
+|**Issue Number**|**Issue Description**|
+|----|----|
+| PWX-13189 | A recursive `chmod` operation in Kubernetes 1.18 and lower caused mounts with large block volumes to hang when users set a security context for a pod using `fsGroup`. <br/><br/>**User Impact:** Users with hung mounts would see mount timeouts when creating a pod referencing large block volumes, and pod creation would fail. <br/><br/>**Resolution:** Portworx, Inc. added the `allow_others` storage class label that, when set to true, will apply a permission change to the mount path. This label should only be used for Kubernetes versions lower than 1.18. Users on newer Kubernetes versions can return to using `fsGroup` over the Portworx `allow_others=true` label. | 
+| PWX-12827 | When running lengthy `pxctl` operations, Portworx timed-out and restarted. <br/><br/>**User Impact:** Users were unable to successfully run `pxctl` operations if they took too long to execute<br/><br/>**Resolution:** Execution timeout can be increased through `pxctl` cluster options. |
```
Comment (vilasd) [inline content/reference/release-notes/portworx.md:33]: Take this out from release note

File: content/reference/release-notes/portworx.md [hunk 8]
```diff
@@ -6,6 +6,36 @@ keywords: portworx, release notes
...
+
+| **Improvement Number** | **Improvement Description** |
+|----|----|
+| PWX-12737 | Added support for automatic cloud disk management for non-public Azure clouds like US Government, Germany and China clouds. |
+| PWX-13082 | You can now configure the frequency with which Portworx takes its KVDB backups using the `kvdb_dump_frequency_minutes` runtime option. |
+| PWX-10216 | Added support for Vault Namespaces. |
+| PWX-11288 | Added support for Vault Kubernetes Authentication. |
+| PWX-12603 | When sharedv4 volumes are in use, Portworx uses 16 NFS threads to process them by default. You can change the total number of threads Portworx uses by running the `pxctl cluster options update --sharedv4-threads <num>` command. |
+| PWX-12512 | PX-Essential can now refresh licenses through an HTTP or HTTPS proxy. |
+| PWX-13116 | Users can now request max backups to be enumerated anywhere between > 0 and < 5000. |
```
Comment (nathan-px) [inline content/reference/release-notes/portworx.md:25]: @vilasd I kept it simple and left out the reasoning behind limiting to 5000. Do we need that information?

**Pull request merged**